### PR TITLE
MathExtras/test: increase coverage

### PR DESCRIPTION
**Pull request by artagnon**
Increase test coverage, and cover possible overflow cases in preparation for another patch optimizing for bitwidth.

### CODE REVIEW

File: llvm/unittests/Support/MathExtrasTest.cpp [hunk 1]
```diff
@@ -183,15 +185,34 @@ TEST(MathExtras, NextPowerOf2) {
   EXPECT_EQ(256u, NextPowerOf2(128));
 }
 
-TEST(MathExtras, alignTo) {
+TEST(MathExtras, AlignTo) {
   EXPECT_EQ(8u, alignTo(5, 8));
   EXPECT_EQ(24u, alignTo(17, 8));
   EXPECT_EQ(0u, alignTo(~0LL, 8));
```
Comment (feg208) [inline llvm/unittests/Support/MathExtrasTest.cpp:191]: this is somewhat a llvm style thing but do we prefer reinterpret/static_cast to c-style casts?

**Review by feg208 — COMMENTED** [review]
seems like some good additional test cases. I had a question about casting but otherwise lgtm. A domain expert is probably required here though

File: llvm/unittests/Support/MathExtrasTest.cpp (hunk 1 above)
Comment (artagnon) [reply llvm/unittests/Support/MathExtrasTest.cpp:191]: We do, in general, but this particular case is just a simple cast from one integer type to another, preserving signedness, so I don't think it matters.

Comment (arsenm) [reply llvm/unittests/Support/MathExtrasTest.cpp:191]: Yes should be static_cast

**Review by nikic — APPROVED** [review]

**Pull request merged**